### PR TITLE
feat: Handle TCU41_x04 as gateway

### PR DESCRIPTION
**Pull request by CFenner**

### CODE REVIEW

**Comment by CFenner** [comment]
@xSauron back then in https://github.com/home-assistant/core/issues/122056 we ignore your gateway directly in HA. I now would like to handle that in the library correctly. For that I would need to know the *roles* that are applied to your gateway. Could you provide that by doing an API call via `curl`?

```
curl -X GET --url https://api.viessmann.com/iot/v2/features/installations/xxx/gateways/xxx/features
```

**Comment by xSauron** [comment]
Hey, sorry for late response - sending the body from `https://api.viessmann.com/iot/v1/equipment/installations` regarding gateways.

```
"gateways": [
{
	"serial": "X",
	"version": "504.2244.48.0",
	"firmwareUpdateFailureCounter": 0,
	"autoUpdate": false,
	"createdAt": "2023-05-23T14:33:22.586Z",
	"producedAt": "2023-05-23T14:33:22.345Z",
	"lastStatusChangedAt": "2024-10-03T03:39:16.256Z",
	"aggregatedStatus": "WorksProperly",
	"targetRealm": "DC",
	"gatewayType": "WiFi_SA0041",
	"installationId": X,
	"registeredAt": "2023-12-22T16:53:44.264Z",
	"description": null,
	"otaOngoing": false,
	"devices": [
		{
			"gatewaySerial": "X",
			"id": "TCU",
			"boilerSerial": null,
			"boilerSerialEditor": null,
			"bmuSerial": null,
			"bmuSerialEditor": null,
			"createdAt": "2023-12-22T16:48:25.363Z",
			"editedAt": "2024-10-03T03:38:42.434Z",
			"modelId": "E3_TCU41_x04",
			"status": "Online",
			"deviceType": "tcu",
			"roles": [
				"capability:zigbeeCoordinator",
				"type:E3",
				"type:gateway;TCU100"
			],
			"isBoilerSerialEditable": false
		},
		{
			"gatewaySerial": "X",
			"id": "0",
			"boilerSerial": "X",
			"boilerSerialEditor": "DeviceCommunication",
			"bmuSerial": null,
			"bmuSerialEditor": null,
			"createdAt": "2023-12-22T16:48:25.363Z",
			"editedAt": "2024-10-03T03:38:42.434Z",
			"modelId": "E3_Vitodens_100_BHC_0421",
			"status": "Online",
			"deviceType": "heating",
			"roles": [
				"capability:backup;0037_HBMU_VD",
				"capability:consumptionReport;electric",
				"capability:consumptionReport;thermal",
				"capability:monetization;AdvancedReport",
				"capability:productionReport;electric",
				"capability:productionReport;thermal",
				"type:E3",
				"type:boiler",
				"type:dhw;integrated",
				"type:heating;integrated",
				"type:product;Vitodens_100"
			],
			"isBoilerSerialEditable": false
		}
	]
}
]
```
Here from `https://api.viessmann.com/iot/v2/features/installations/xxx/gateways/xxx/features`

```
"properties": {
  "devices": {
	  "type": "DeviceList",
	  "value": [
		  {
			  "id": "0",
			  "fingerprint": "X",
			  "modelId": "E3_Vitodens_100_BHC_0421",
			  "modelVersion": "b9MCJ6Sa2zUrIc7Lyd1-jcf3CB4",
			  "name": "Vitodens One Hundred E3, Release 0620, MCD-3D version",
			  "type": "heating",
			  "roles": [
				  "capability:backup;0037_HBMU_VD",
				  "capability:consumptionReport;electric",
				  "capability:consumptionReport;thermal",
				  "capability:monetization;AdvancedReport",
				  "capability:productionReport;electric",
				  "capability:productionReport;thermal",
				  "type:E3",
				  "type:boiler",
				  "type:dhw;integrated",
				  "type:heating;integrated",
				  "type:product;Vitodens_100"
			  ],
			  "status": "online"
		  },
		  {
			  "id": "TCU",
			  "fingerprint": "X",
			  "modelId": "E3_TCU41_x04",
			  "modelVersion": "34nbqQJIrKwwVGAhx6H6sjqLUI8",
			  "name": "TCU",
			  "type": "tcu",
			  "roles": [
				  "capability:zigbeeCoordinator",
				  "type:E3",
				  "type:gateway;TCU100"
			  ],
			  "status": "online"
		  }
	  ]
  }
  },
```

File: PyViCare/PyViCareDeviceConfig.py [hunk 1]
```diff
@@ -88,6 +88,7 @@ def asAutoDetectDevice(self):
             (self.asRoomSensor, r"E3_RoomSensor", ["type:climateSensor"]),
             (self.asGateway, r"E3_TCU19_x05", ["type:gateway;TCU200"]),
             (self.asGateway, r"E3_TCU10_x07", ["type:gateway;TCU300"]),
+            (self.asGateway, r"E3_TCU41_x04", ["type:gateway;TCU???"]),
```
Comment (CFenner) [inline PyViCare/PyViCareDeviceConfig.py]:
```suggestion
            (self.asGateway, r"E3_TCU41_x04", ["type:gateway;TCU100"]),
```

File: PyViCare/PyViCareService.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def hasRoles(self, requested_roles) -> bool:
         return hasRoles(requested_roles, self.roles)
 
     def _isGateway(self) -> bool:
-        return self.hasRoles(["type:gateway;VitoconnectOpto1"]) or self.hasRoles(["type:gateway;VitoconnectOpto2/OT2"]) or self.hasRoles(["type:gateway;TCU300"])
+        return self.hasRoles(["type:gateway;VitoconnectOpto1"]) or self.hasRoles(["type:gateway;VitoconnectOpto2/OT2"]) or self.hasRoles(["type:gateway;TCU200"]) or self.hasRoles(["type:gateway;TCU300"])
```
Comment (CFenner) [inline PyViCare/PyViCareService.py]:
```suggestion
        return self.hasRoles(["type:gateway;VitoconnectOpto1"]) or self.hasRoles(["type:gateway;VitoconnectOpto2/OT2"]) or self.hasRoles(["type:gateway;TCU100"]) or self.hasRoles(["type:gateway;TCU300"])
```

**Pull request merged**